### PR TITLE
Add Regal linter

### PR DESCRIPTION
**Pull request by zregvart**
Integrates the Regal linter with one provision to mark `TODO` comments as a warning instead of a failure.

### CODE REVIEW

File: policy/pipeline/required_tasks_test.rego [hunk 1]
```diff
@@ -21,27 +22,32 @@ test_required_tasks_not_met if {
...
 	pipeline_finally := _pipeline_with_tasks_and_label([], _expected_future_required_tasks, [])
-	lib.assert_empty(warn) with data["pipeline-required-tasks"] as _time_based_pipeline_required_tasks
+	lib.assert_empty(required_tasks.warn) with data["pipeline-required-tasks"] as _time_based_pipeline_required_tasks
 		with input as pipeline_finally
 }
 
 test_not_warn_if_only_future_required_tasks if {
 	pipeline := _pipeline_with_tasks_and_label(_expected_future_required_tasks, [], [])
-	lib.assert_empty(warn) with data["pipeline-required-tasks"] as _time_based_pipeline_required_tasks_future_only
+	tasks1 = _time_based_pipeline_required_tasks_future_only
```
Comment (lcarva) [inline policy/pipeline/required_tasks_test.rego:44]: I'm surprised it doesn't want: `tasks1 := ...`

Wait, why have `tasks` and `tasks2`? Can't we use the same in both cases?

File: policy/release/olm.rego [hunk 2]
```diff
@@ -32,6 +32,7 @@ olm_manifestv1 := "operators.operatorframework.io.bundle.manifests.v1"
 #   - redhat
 #
 deny contains result if {
+	# regal ignore:prefer-snake-case
 	manifestDir := input.image.config.Labels[olm_manifestv1]
```
Comment (lcarva) [inline policy/release/olm.rego:36]: For this one, we should probably just use `manifest_dir`.

File: policy/lib/bundles.rego [hunk 3]
```diff
@@ -7,57 +7,48 @@ import data.lib.refs
...
 
 # Returns a subset of tasks that do not use a bundle reference.
-disallowed_task_reference(tasks) = matches {
-	matches := {task |
-		task := tasks[_]
-		not bundle(task)
-	}
+disallowed_task_reference(tasks) := {task |
+	some task in tasks
+	not refs.task_ref(task).bundle
```
[the remark this answers is not in the document]
Comment (zregvart) [reply policy/lib/bundles.rego:12]: The rule fails in that case, `bundle(task)` is undefined so `disallowed_task_reference` becomes undefined

File: policy/pipeline/required_tasks_test.rego (hunk 1 above)
Comment (zregvart) [reply policy/pipeline/required_tasks_test.rego:44]: Been staring at this for too long I guess...

File: policy/lib/tekton/task.rego [hunk 4]
```diff
@@ -132,8 +124,10 @@ _task_params(task) := params if {
 }
 
 # task_param returns the value of the given parameter in the task.
-task_param(task, name) := value if {
-	value := _task_params(task)[name]
+task_param(task, name) := params if {
+	# without this we end up with https://docs.styra.com/regal/rules/bugs/top-level-iteration
```
[the remark this answers is not in the document]
Comment (zregvart) [reply policy/lib/tekton/task.rego:128]: Filled https://github.com/StyraInc/regal/pull/347, lets have this as is until that change is accepted

[the remark this answers is not in the document]
Comment (lcarva) [reply policy/lib/tekton/task.rego:128]: Works for me! It's merged now. Do we still this until the next release, maybe?

File: policy/release/sbom_spdx_test.rego [hunk 5]
```diff
@@ -78,7 +78,7 @@ test_missing_packages if {
 		with input.image.ref as "registry.local/spam@sha256:123"
 }
 
-test_missing_packages if {
+test_missing_files if {
```
Comment (lcarva) [inline policy/release/sbom_spdx_test.rego:81]: :facepalm: 

Thanks for fixing this!

**Review by lcarva — APPROVED** [review]

**Comment by lcarva** [comment]
I'm gonna merge this now to prevent it getting out of date due to the large number of files changes.

**Pull request merged**